app/models/king.rb

```diff
@@ -12,6 +12,39 @@ def valid_move?(x, y)
     return false if blocked?(x, y) # checks if piece is blocking at destination coordinate
     return false if delta_x(x_position, x) > 1
     return false if delta_y(y_position, y) > 1
+
+    # The king can castle
+    return true if move_castle?(x, y)
+
     true
   end
+
+  # Support castling with the king
+  def move_castle?(x, y)
+    castling_kingside?(x, y) && castling_never_moved?
+  end
+
+  # The king can attack to the castle on the king side
+  def castling_kingside?(x, y)
+    x == 6 && y == y_position && castling_rook_kingside && castling_rook_kingside.castling_never_moved? && \
+      castling_rook_kingside.blocked?(4, y_position) == false
+  end
+
+  # The rook to castle with king side
+  def castling_rook_kingside
+    can_castling_rook_kingside = game.pieces.where(x_position: 0, y_position: y_position, \
```

Comment:
I think you want to grab the rook that has the same color as the current king, so you can change `color: 'white' || 'black'` to just `color: color`.